src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -721,7 +736,7 @@ private String computeEnclosingLabel(FlowNode executorStepNode, List<FlowNode> h
         }
 
         @Override public long getEstimatedDuration() {
-            Run<?,?> r = run();
+            Run<?,?> r = runForDisplay();
```

Comment:
Also only tangentially related—there was no apparent reason not to be calling the newer and more robust `runForDisplay` in this context.